src/client/windshaft.js

```diff
@@ -320,25 +331,49 @@ export default class Windshaft {
         }
         return {
             urlTemplates: layergroup.metadata.tilejson.vector.tiles,
-            metadata: overrideMetadata || this._adaptMetadata(layergroup.metadata.layers[0].meta, agg)
+            metadata: overrideMetadata || this._adaptMetadata(layergroup.metadata.layers[0].meta, agg, MNS)
         };
     }
 
-    _adaptMetadata (meta, agg) {
+    _adaptMetadata (meta, agg, MNS) {
         meta.datesAsNumbers = meta.dates_as_numbers;
         const { stats, aggregation, datesAsNumbers } = meta;
         const featureCount = stats.hasOwnProperty('featureCount') ? stats.featureCount : stats.estimatedFeatureCount;
         const geomType = stats.geometryType && adaptGeometryType(stats.geometryType);
 
         const properties = stats.columns;
         Object.keys(agg.columns).forEach(aggName => {
```

Comment:
We could extract both Object.keys blocks to little private functions, to simplify a bit this method, something like:
`_adaptAggregationColumns` and `_adaptAggregationDimensions`

Reply:
You mean something like:

```
this._adaptAggregationColumns(aggName => ...
```

Or do you think it would be better just to:

```
this._adaptAggregationColumns().forEach(aggName => ...
```